File: profiles/go/scaffold/health.go
// Package main is a dependency-free (stdlib-only) starter service.
//
// health.go holds ONLY the pure health core; the routing, security middleware,
// per-request telemetry, and socket-binding boot live in server.go (the ONE place
// the profile assembles the flags + telemetry modules into a running server).
// Keeping Health() pure (no I/O) makes it directly unit-testable.
package main

// Health is a pure function returning the service health payload.
// Keeping it pure (no I/O) makes it directly unit-testable and lets both the
// /healthz route (server.go) and the container HEALTHCHECK probe share one source
// of truth for the exact body.
func Health() map[string]string {
	return map[string]string{"status": "ok"}
}

File: profiles/rust/scaffold/src/health.rs
//! Pure health core (Rust profile).
//!
//! Holds ONLY the pure health payload; the routing, security middleware,
//! per-request telemetry, and socket-binding boot live in [`crate::server`] (the ONE
//! place the profile assembles the flags + telemetry modules into a running server).
//! Keeping [`health`] pure (no I/O) makes it directly unit-testable and lets both the
//! `/healthz` route and any container HEALTHCHECK probe share one source of truth for
//! the exact body. Mirrors the go reference's `Health()` and python's health payload.

/// The service health payload as a compact JSON string. Pure — no I/O — so it is
/// directly testable and is the single source of truth for the `/healthz` body.
#[must_use]
pub fn health() -> &'static str {
    "{\"status\":\"ok\"}"
}

#[cfg(test)]
mod tests {
    use super::health;

    #[test]
    fn health_is_ok() {
        assert_eq!(health(), "{\"status\":\"ok\"}");
    }
}
